File: 643-maximum-average-subarray-i/643-maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) 
    {
      int n = nums.size();  
      double max_val =INT_MIN;  
        
            int p1=1;
            int sum =0;
            //int p2 = p1+k-1;
            
            for(int i=0;i<k;++i)
               sum+=nums[i];
                
            max_val = max(max_val,(double)sum/k);
            
             while(p1+k<=nums.size())
            {
                sum-=nums[p1-1];
                sum+=nums[p1+k-1];
                max_val = max(max_val,(double)sum/k);
                p1++;
             }     
                
        return max_val;
    }
};

File: 643-maximum-average-subarray-i/643-maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) 
    {
      double max=Integer.MIN_VALUE;
        double sum=0;
        for(int i=0;i<k;i++)
        {
            sum+=nums[i];
        }
        max=sum/k; 
        int last=k;
        for(int start=0;last<nums.length;start++)
        {
            double average=0;
            sum=sum-nums[start];
            sum=sum+nums[last];
            last++;
            average=sum/k;
            if(max<average)
                max=average;
        }
        return max;
    }
}